memory_core/memtile_util.py

```diff
@@ -90,10 +91,10 @@ def wrap_lake_core(self):
                 app_list = self.__outputs
             if ind_ports > 1:
                 for i in range(ind_ports):
-                    self.add_port(f"{io_info.port_name}_{i}", dir_type(intf_type))
+                    self.io += m.IO(f"{io_info.port_name}_{i}"=dir_type(intf_type))
```

Comment:
> `f"{io_info.port_name}_{i}"=dir_type(intf_type)`

This works!?? It looks like you're assigning a value to a string constant...??

Reply:
No, it's creating a new port. That's a key/value pair, where the key is the name of the new port, and the value is the type of the port. 

It is true that it doesn't work though... @rsetaluri any suggestions on how to create a port using an expression for the name, rather than just directly writing the port name? Is this possible?

Reply:
It did not work, but it should now after my recent push. I got confused and thought I was creating a dictionary with string keys, but these are actually kwargs. See the recent commits for the fix.